src/Xamarin.Android.Build.Tasks/Tasks/GetAotArguments.cs

```diff
@@ -291,19 +292,27 @@ string GetLdFlags(NdkTools ndk, AndroidTargetArch arch, int level, string toolPr
 				libs.Add (Path.Combine (androidLibPath, "libc.so"));
 				libs.Add (Path.Combine (androidLibPath, "libm.so"));
 
-				ldFlags = $"\\\"{string.Join ("\\\";\\\"", libs)}\\\"";
+				ldFlags.Append ($"\\\"{string.Join ("\\\";\\\"", libs)}\\\"");
 			}
 
-			if (!StripLibraries) {
-				return ldFlags;
+			if (ldFlags.Length > 0) {
+				ldFlags.Append (' ');
 			}
 
-			const string StripFlag = "-s";
-			if (ldFlags.Length == 0) {
-				return StripFlag;
+			//
+			// This flag is needed for Mono AOT to work correctly with the LLVM 14 `lld` linker due to the following change:
+			//
+			//   The AArch64 port now supports adrp+ldr and adrp+add optimizations. --no-relax can suppress the optimization.
+			//
+			// Without the flag, `lld` will modify AOT-generated code in a way that the Mono runtime doesn't support. Until
+			// the runtime issue is fixed, we need to pass this flag then.
+			//
+			ldFlags.Append ("--no-relax");
```

Comment:
Do we need an MSBuild property to turn this off? It could be private like `$(_AndroidAotNoRelax)` that defaults to `true`.

Reply:
@jonathanpeppers I wouldn't bother. We can just remove the code when we move to fixed Mono AOT